src/services/logsFrame.ts

```diff
@@ -180,3 +190,41 @@ export function getBodyName(logsFrame?: LogsFrame | null): string {
 export function getIdName(logsFrame?: LogsFrame): string {
   return logsFrame?.idField?.name ?? DATAPLANE_ID_NAME;
 }
+
+export function getSeriesVisibleRange(series: DataFrame[]) {
+  let start = 0;
+  let end = 0;
+
+  const timeField = series[0]?.fields.find((field) => field.type === FieldType.time);
+  if (timeField) {
+    const values = timeField.values.sort();
+    const oldestFirst = values[0] < values[values.length - 1];
+    start = oldestFirst ? values[0] : values[values.length - 1];
+    end = oldestFirst ? values[values.length - 1] : values[0];
+  }
+  return { start, end };
+}
+
+export const VISIBLE_RANGE_NAME = 'Visible range';
+export function getVisibleRangeFrame(start: number, end: number) {
+  const frame = arrayToDataFrame([
+    {
+      time: start,
+      timeEnd: start,
+      text: 'Time of the oldest log',
```

Comment:
Another nit: can we make the copy a tad more specific to this feature? 
`Oldest log rendered in logs panel`?


<img width="469" alt="image" src="https://github.com/user-attachments/assets/7f7636f8-6ca8-4a8b-ab8e-b290d5c1786a">

Maybe create an issue to add a button to the annotation region tooltip that scrolls to that specific log?

Reply:
Definitely. Using a painted region, how does this sound? 

![imagen](https://github.com/user-attachments/assets/acb2bd4a-17d4-48e7-9d6c-e67f9ca88bb6)

> Maybe create an issue to add a button to the annotation region tooltip that scrolls to that specific log?

Created https://github.com/grafana/grafana/issues/97257